homeassistant/components/nws/__init__.py

```diff
@@ -58,36 +59,55 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     nws_data = SimpleNWS(latitude, longitude, api_key, client_session)
     await nws_data.set_station(station)
 
-    async def update_observation() -> None:
-        """Retrieve recent observations."""
-        await call_with_retry(
-            nws_data.update_observation,
-            RETRY_INTERVAL,
-            RETRY_STOP,
-            start_time=utcnow() - UPDATE_TIME_PERIOD,
-        )
-
-    async def update_forecast() -> None:
-        """Retrieve twice-daily forecsat."""
-        await call_with_retry(
-            nws_data.update_forecast,
-            RETRY_INTERVAL,
-            RETRY_STOP,
-        )
-
-    async def update_forecast_hourly() -> None:
-        """Retrieve hourly forecast."""
-        await call_with_retry(
-            nws_data.update_forecast_hourly,
-            RETRY_INTERVAL,
-            RETRY_STOP,
-        )
-
+    def setup_update_observation(
```

Comment:
```suggestion
    def async_setup_update_observation(
```

Since we have both sync and async code in this code base please prefix functions that are safe to run in the event loop with `async_`

https://developers.home-assistant.io/docs/asyncio_categorizing_functions?_highlight=callback#the-callback-function

Reply:
Should I use `@callback` here as it isn't a 'callback' or just the naming convention?